low-level/framework/platforms/server/hba.py

```diff
@@ -0,0 +1,147 @@
+
+        # TODO: Virtual memory may get exhausted by running scan command.
+        # Scanning may take long time. To avoid sensor polling overlapping,
+        # scan finish time must be known. HBA Sensor polling interval must
+        # be greater than scan finish time.
+        # Rescan multipath
+        #self._rescan_scsi_bus()
+
+
+    @staticmethod
+    def _rescan_scsi_bus():
+        """Scan and detect new LUNs."""
+
+        _, err, rc = SimpleProcess("rescan-scsi-bus.sh").run()
+        if rc != 0:
+            raise HBAError(rc, "Failed to scan scsi bus due to %s", err)
+
+
+    def _is_host_found(self, host):
+        """
+        Check host is present.
+
+        Parameters:
+            host (str): scsi/fc host
+        """
+
+        if host not in self.get_hosts():
+            return False
+        return True
+
+
+    def _get_scsi_host_data(self, host):
+        """
+        Returns SCSI host information.
+
+        Parameters:
+            host (str): scsi host detected by HBA card
+        Returns:
+            data (dict): scsi host information
+        """
+
+        data = {}
+
+        if self._is_host_found(host):
+            data = {
+                "port_id": self.__utility.get_host_data(host, 'unique_id'),
+                "state": self.__utility.get_host_data(host, 'state'),
+                "active_mode": self.__utility.get_host_data(host, 'active_mode'),
+                "host_busy": self.__utility.get_host_data(host, 'host_busy')
+                }
+
+        return data
+
+
+    def _get_fc_host_data(self, host):
+        """
+        Returns FC host information.
+
+        Parameters:
+            host (str): scsi host detected by HBA card
+        Returns:
+            data (dict): fc host information
+        """
+
+        data = {}
+
+        if self._is_host_found(host):
+            data = {
+                "port_id": self.__utility.get_host_data(host, 'port_name'),
+                "state": self.__utility.get_host_data(host, 'port_state')
+                }
+
+        return data
+
+
+    def initialize_utility(self, utility):
+        """Invoke HBA card utility tool."""
+
+        self.__utility = ToolFactory().get_instance(utility)
+
+
+    def get_hosts(self):
+        """Returns hosts detected by HBA card."""
+
+        hosts = self.__utility.get_host_list()
+
+        # Refresh host type
+        self.host_type = self.__utility.host_type
+        self.detected = True if self.host_type else False
+
+        return hosts
+
+
+    def get_host_data(self, host):
+        """
+        Returns host information based on its type.
+
+        Parameters:
+            host (str): scsi host detected by HBA card
+        Returns:
+            data (dict): fc host information
+        """
+
+        data = {}
+
+        if self.host_type == 'scsi_host':
+            data = self._get_scsi_host_data(host)
+        elif self.host_type == 'fc_host':
+            data = self._get_fc_host_data(host)
+
```

Comment:
Can we make it host type as Enum class in common file and import that ? instead of hardcode.

Reply:
Rather than Enum, referring these from sspl_constants.py would be suffice. Let me know if you have any second thought.

```sspl_constants.py

FC_HOST = "fc_host"
SCSI_HOST = "scsi_host"```

Reply:
Done.